### DISCUSSION TITLE
Need helps on how to build a pandas dataframe including these items from listing.

### DISCUSSION THREAD

**Discussion by Mark00353** [AnthonyBloomer/daftlistings#142] [Q&A] [answered]
Hi, I am a green hand here, and thanks a million for the daftlistings codes:

for listing in listings:
    print(listing.publish_date)
    print(listing.title)
    print(listing.price)

I attempted to list these items into a pandas dataframe layout and export the results into an excel format, somehow I failed to do so, Can anyone helps me on this matter?

Thanks again.

**Comment by sdfordham** [comment, marked as answer]
If you want something openable with excel, then use the `csv` package from the standard library. Make sure to open with `encoding='utf-8'` or the euro symbol will get mangled.

````python
import csv
from daftlistings import Daft, Location, SearchType, PropertyType

daft = Daft()
daft.set_location(Location.DUBLIN)
daft.set_search_type(SearchType.RESIDENTIAL_RENT)
daft.set_property_type(PropertyType.APARTMENT)
daft.set_max_price(700)

listings = daft.search()

with open('./listings.csv', 'w', newline='', encoding='utf-8') as F:
    writer = csv.writer(F)
    writer.writerow(['title', 'price', 'link']) # Column names
    for listing in listings:
        writer.writerow([listing.title, listing.price, listing.daft_link])
````
If you really want the listings in a `pandas.DataFrame` then there are many ways to skin this cat, I prefer to create from a dictionary.
````python
import pandas as pd
from daftlistings import Daft, Location, SearchType, PropertyType

daft = Daft()
daft.set_location(Location.DUBLIN)
daft.set_search_type(SearchType.RESIDENTIAL_RENT)
daft.set_property_type(PropertyType.APARTMENT)
daft.set_max_price(700)

listings = daft.search()

# Convert to a list of dictionaries
listings = [
    {'title': listing.title,
     'price': listing.price,
     'link': listing.daft_link} for listing in listings
]
df = pd.DataFrame.from_dict(listings)
print(df)
````

**Reply by Mark00353** [reply]
Thanks a million Sir!

Both options you mentioned at above are extremely helpful, I needed to build the the dataframe for plotting (like matplotlib) and also needed them in CSV format to play around with them.

Thanks again.
